src/leveled_sst.erl

```diff
@@ -895,41 +885,75 @@ delete_pending({get_kv, LedgerKey, Hash}, _From, State) ->
             State#state.handle,
             State#state.level,
             {no_monitor, 0}),
-    {reply, Result, delete_pending, State, ?DELETE_TIMEOUT};
-delete_pending({get_kvrange, StartKey, EndKey, ScanWidth, SegList, LowLastMod},
-                                                            _From, State) ->
+   {keep_state_and_data,
+    [{reply, From, Result}, ?DELETE_TIMEOUT]};
+delete_pending({call, From}, {get_kvrange, StartKey, EndKey, ScanWidth, SegList, LowLastMod},
+               State) ->
     {_NeedBlockIdx, SlotsToFetchBinList, SlotsToPoint} =
         fetch_range(StartKey, EndKey, ScanWidth, SegList, LowLastMod, State),
     % Always yield as about to clear and de-reference
     PressMethod = State#state.compression_method,
     IdxModDate = State#state.index_moddate,
-    {reply,
-        {yield, SlotsToFetchBinList, SlotsToPoint, PressMethod, IdxModDate},
-        delete_pending,
-        State,
-        ?DELETE_TIMEOUT};
-delete_pending({get_slots, SlotList, SegList, LowLastMod}, _From, State) ->
+    {keep_state_and_data,
+    [{reply, From,
+        {yield, SlotsToFetchBinList, SlotsToPoint, PressMethod, IdxModDate}},
+     ?DELETE_TIMEOUT]};
+delete_pending({call, From}, {get_slots, SlotList, SegList, LowLastMod}, State) ->
     PressMethod = State#state.compression_method,
     IdxModDate = State#state.index_moddate,
-    {_NeedBlockIdx, SlotBins} = 
-        read_slots(State#state.handle, 
-                    SlotList, 
+    {_NeedBlockIdx, SlotBins} =
+        read_slots(State#state.handle,
+                    SlotList,
                     {SegList, LowLastMod, State#state.blockindex_cache},
                     PressMethod,
                     IdxModDate),
-    {reply, 
-        {false, SlotBins, PressMethod, IdxModDate}, 
-        delete_pending, 
-        State, 
-        ?DELETE_TIMEOUT};
-delete_pending(close, _From, State) ->
+    {keep_state_and_data,
+    [{reply, From,
+        {false, SlotBins, PressMethod, IdxModDate}},
+      ?DELETE_TIMEOUT]};
+delete_pending({call, From}, close, State) ->
+    leveled_log:log(sst07, [State#state.filename]),
+    ok = file:close(State#state.handle),
+    ok = file:delete(filename:join(State#state.root_path,
+                                    State#state.filename)),
+    {stop_and_reply, normal, [{reply, From, ok}], State};
+delete_pending(cast, close, State) ->
     leveled_log:log(sst07, [State#state.filename]),
     ok = file:close(State#state.handle),
     ok = file:delete(filename:join(State#state.root_path,
                                     State#state.filename)),
-    {stop, normal, ok, State}.
+    {stop, normal, State};
+
+delete_pending(cast, {update_blockindex_cache, BIC}, State) ->
+    handle_update_blockindex_cache(BIC, State);
```

Comment:
This is interesting. get_slots may be prompted which results in update_blockindex_cache but a set_for_delete message may happen in-between.  Hence we do need support for update_blockindex_cache in delete_pending state - but this is never triggered in test, so we perhaps need a  unit test to explicitly trigger it.